src/controllers/cart.controller.ts

```diff
@@ -127,6 +127,70 @@ export const cartController = {
     } catch (error) {
       res.status(500).json({ message: 'Internal server error' });
     }
+  },
+
+  async clearCart(req: CustomRequest, res:Response) {
+    try {
+      const userId = req.user;
+      if(!userId){
+        return res.status(401).json({ message: 'Unauthorized' });
+      }

```

Comment:
Isn't the middleware already doing this? if not please make sure it is and remove this from here
```suggestion
```